lib/transports.js

```diff
@@ -25,6 +27,17 @@ HTTPTransport.prototype.send = function(client, message, headers, eventId, cb) {
     port: client.dsn.port || this.defaultPort,
     ca: client.ca
   };
+
+  // set path apprpriately when using proxy
+  if (this.options.hasOwnProperty('host')) {
```

Comment:
@ndmanvar I think you're missing setting the `Host` header to be set to `client.dsn.host` in case of proxy.
Without it the host header would keep the proxy host which cause the HTTP request to be rejected by the sentry.io server